gensim/models/keyedvectors.py

```diff
@@ -617,6 +618,22 @@ def similarity(self, w1, w2):
 
         """
         return dot(matutils.unitvec(self[w1]), matutils.unitvec(self[w2]))
+    
+    def most_similar_to_given(self, w1, word_list):
+        """
+        Return the word from word_list most similar to w1.
+        
+        Example::
+            
+          >>> trained_model.most_similar_to_given('music', ['water', 'sound', 'backpack', 'mouse'])
```

Comment:
What do you this @gojomo, it's a useful feature?

Reply:
If done efficiently it makes sense. If the common case is there's a stable subset against which some projects are doing all of their similarity-searches, the need might be best met by a new method for creating a subset KeyedVectors, with just the words-of-interest. Overlaps with closed-as-idle PR #1229.